### PR TITLE
Documenting Collection::through()

### PR DESCRIPTION
**Pull request by lorenzo**

### CODE REVIEW

**Comment by markstory** [comment]
My comments were added as a commit.

File: en/core-libraries/collections.rst [hunk 1]
```diff
@@ -721,6 +735,94 @@ first one, then the target property will be filled with ``null`` values::
...
+and focused, but it can get messy very quickly. This becomes more obvious when
+a lot of different methods need to be called or when the length of the closure
+methods is more than just a few lines.
+
+There are also cases when the logic used for the collection methods can be
+reused in multiple parts of your application. It is recommended that you
+consider extracting complex collection logic to separate classes. For example,
+imagine a lengthy closure like this one::
+
+        $colletion
```
Comment (dereuromark) [inline en/core-libraries/collections.rst]: Typo collection

**Comment by antograssiot** [comment]
@lorenzo I've updated some code samples after testing it

**Comment by lorenzo** [comment]
Thanks for your comments and fixes :)

**Pull request merged**